File: src/contests/cg/puzzle/pascal/the-river-i-.pas
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

{$mode objfpc}{$H+}
var
  a, b: int64;

function digitSum(x: int64): int64;
var
  s: int64;
begin
  s := 0;
  while x > 0 do
  begin
    s := s + x mod 10;
    x := x div 10;
  end;
  digitSum := s;
end;

begin
  readln(a);
  readln(b);
  while a <> b do
    if a < b then a := a + digitSum(a) else b := b + digitSum(b);
  writeln(a);
end.

File: src/contests/cg/puzzle/ocaml/the-river-i-.ml
(* 🎮 CodinGame Puzzle - the-river-i- *)
(* https://www.codingame.com/training/easy/the-river-i- *)

let digit_sum x =
  let x = ref x and s = ref 0 in
  while !x > 0 do
    s := !s + !x mod 10;
    x := !x / 10
  done;
  !s

let () =
  let a = ref (int_of_string (String.trim (read_line ()))) in
  let b = ref (int_of_string (String.trim (read_line ()))) in
  while !a <> !b do
    if !a < !b then a := !a + digit_sum !a
    else b := !b + digit_sum !b
  done;
  Printf.printf "%d\n" !a
